libfwupdplugin/fu-quirks.c

```diff
@@ -146,6 +146,19 @@ fu_quirks_convert_quirk_to_xml_cb(XbBuilderSource *source,
 			value = g_key_file_get_value(kf, groups[i], keys[j], error);
 			if (value == NULL)
 				return NULL;
+
+			/* sanity check flags */
+			if (g_strcmp0(keys[j], FU_QUIRKS_FLAGS) == 0) {
+				g_autofree gchar *value_down = g_utf8_strdown(value, -1);
+				if (g_strcmp0(value, value_down) != 0 ||
+				    g_strstr_len(value, -1, " ") != NULL) {
```

Comment:
Would `g_ascii_isalnum` be a better check here?

Reply:
I've added something more comprehensive.